azkaban-common/src/main/java/azkaban/utils/KafkaLog4jUtils.java

```diff
@@ -0,0 +1,94 @@
+package azkaban.utils;
+
+import static azkaban.Constants.ConfigurationKeys.AZKABAN_LOGGING_KAFKA_CLASS_PARAM;
+import static azkaban.Constants.ConfigurationKeys.AZKABAN_LOGGING_KAFKA_ENABLED;
+
+import azkaban.Constants;
+import azkaban.Constants.ConfigurationKeys;
+import java.lang.reflect.Constructor;
+import java.util.Arrays;
+import java.util.Objects;
+import org.apache.kafka.log4jappender.KafkaLog4jAppender;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Utility class for getting Azkaban Kafka Log4j Appender
+ */
+public class KafkaLog4jUtils {
+
+  private static final Logger logger = LoggerFactory.getLogger(KafkaLog4jUtils.class);
+
+  private static String getAzkabanKafkaLog4jName(final String[] ids) {
+    return String.join(".", ids);
+  }
+
+  private static KafkaLog4jAppender getAzkabanKafkaLog4jAppender(final Props props,
+      final String execId, final String flowId, final String jobId, final String jobAttempt,
+      final String topicConfigKey) {
+    final boolean loggingKafkaEnabled =
+        props.getBoolean(AZKABAN_LOGGING_KAFKA_ENABLED, false);
+
+    if (!loggingKafkaEnabled) {
+      logger.info("Server logging kafka is not enabled");
+      return null;
+    }
+
+    try {
+      final Class<?> kafkaLog4jAppenderClass =
+          props.getClass(AZKABAN_LOGGING_KAFKA_CLASS_PARAM, KafkaLog4jAppender.class);
+      logger.info("Loading kafka log4j appender class " + kafkaLog4jAppenderClass.getName());
+
+      final Constructor<?> kafkaLog4jAppenderClassConstructor =
+          kafkaLog4jAppenderClass.getConstructor();
+      KafkaLog4jAppender kafkaLog4jAppender =
+          (KafkaLog4jAppender) kafkaLog4jAppenderClassConstructor.newInstance();
+
+      String[] origIds = new String[] {execId, flowId, jobId, jobAttempt};
+      String[] nonNullIds = Arrays.stream(origIds).filter(Objects::nonNull).toArray(String[]::new);
+
+      kafkaLog4jAppender.setBrokerList(props
```

Comment:
It might be better to provide a default value.

Reply:
We cannot give a default value b/c brokerList is the key to kafka (can be treated as the kafka cluster name; we shouldn't give an arbitrary value here).
If no value is given in config, it will fallback to the catch block and return null which will be handled nicely in FlowRunner and JobRunner.